crates/oxc_ast/src/trivia.rs

```diff
@@ -29,9 +34,24 @@ impl Comment {
     pub fn new(end: u32, kind: CommentKind) -> Self {
         Self { kind, end }
     }
+
+    #[must_use]
+    pub fn end(&self) -> u32 {
+        self.end
+    }
 }
 
 impl Trivias {
+    #[must_use]
+    pub fn comments(&self) -> &BTreeMap<u32, Comment> {
+        &self.comments
+    }
+
+    #[must_use]
+    pub fn newlines(&self) -> &BTreeSet<u32> {
+        &self.newlines
+    }
+
```

Comment:
Nit: It may make sense to wrap these types in new-types wrapper to avoid that the `BTreeMap` leaks into public api (you may want to change it in the future).